### PR TITLE
Adopt commitlint standards

### PR DESCRIPTION
**Pull request by tisoap**
## Description
Looking for issues to contribute for Hacktoberfest, I stumbled upon this repository  :slightly_smiling_face:

This PR adds [commitlint](https://commitlint.js.org/#/) standards for commit messages to the project. Fixes issue #27.

New developer dependencies: 
- `@commitlint/cli`
- `@commitlint/config-conventional`

Updated developer dependencies:
- `husky`

I have opted to link the default naming rules documentation instead of describing them inside the project, but if deemed necessary this can be arranged.

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

### CODE REVIEW

File: docs/guide/contributing.md [hunk 1]
```diff
@@ -12,7 +12,8 @@ title: 'Contributing'
 4. Type in `npm link` which creates a symlink and thereby allowing `teachcode` to be accessed globally.
 5. Make your life changing changes.
 6. Ensure that your code is free from linting errors and is as per prettier conventions by typing in `npm run lint`.
-7. Now you may push it to the remote repository by switching over to another branch and finally proposing a Pull Request to the base fork.
+7. We use [commitlint conventional naming rules](https://www.npmjs.com/package/@commitlint/config-conventional#rules) for our commits, make sure to follow them.
```
Comment (jamesgeorge007) [inline docs/guide/contributing.md]:
```suggestion
7. We use [commitlint conventional naming rules](https://www.npmjs.com/package/@commitlint/config-conventional#rules) for our commits, make sure that you follow them.
```

File: docs/guide/contributing.md [hunk 2]
```diff
@@ -12,7 +12,8 @@ title: 'Contributing'
...
+8. Now you may push it to the remote repository by switching over to another branch and finally proposing a Pull Request to the base fork.
```
Comment (jamesgeorge007) [inline docs/guide/contributing.md]: I guess we can also have a mention regarding the branch naming conventions, such as `feat`, `hotfix`, 'docs' prefixes for feature proposals, bug fixes, and docs-related respectively.

Comment (jamesgeorge007) [reply docs/guide/contributing.md]: You may follow the approach as [here](https://github.com/madlabsinc/mevn-cli#how-should-i-write-a-commit-message).

Comment (tisoap) [reply docs/guide/contributing.md]: Just realized there was already some mention to branch naming in the same file, so I expanded the description a bit.

Comment (tisoap) [reply docs/guide/contributing.md]: Do you also want the commit message pattern described in `contributing.md`?

Comment (jamesgeorge007) [reply docs/guide/contributing.md]: Just have a mention of it within `README.md` following [this](https://github.com/madlabsinc/mevn-cli#how-should-i-write-a-commit-message) approach.

Comment (tisoap) [reply docs/guide/contributing.md]: Alright, I've updated this project's `README.md` file to include two new sections under "Contributing": "How should I write a commit message?" and "How should I name my branches?".

**Pull request merged**